gcloud-java-compute/src/main/java/com/google/gcloud/compute/DiskImageConfiguration.java

```diff
@@ -0,0 +1,157 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.services.compute.model.Image;
+import com.google.common.base.MoreObjects;
+
+import java.util.Objects;
+
+/**
+ * A Google Compute Engine disk image configuration. This class can be used to create images from an
+ * existing Google Compute Engine disk.
+ */
+public class DiskImageConfiguration extends ImageConfiguration {
+
+  private static final long serialVersionUID = 2716403667042981170L;
+
+  private final DiskId sourceDisk;
+  private final String sourceDiskId;
+
+  /**
+   * A builder for {@code DiskImageConfiguration} objects.
+   */
+  public static final class Builder
+      extends ImageConfiguration.Builder<DiskImageConfiguration, Builder> {
+
+    private DiskId sourceDisk;
+    private String sourceDiskId;
+
+    private Builder() {}
+
+    private Builder(DiskImageConfiguration imageConfiguration) {
+      super(imageConfiguration);
+      this.sourceDisk = imageConfiguration.sourceDisk;
+      this.sourceDiskId = imageConfiguration.sourceDiskId;
+    }
+
+    private Builder(Image imagePb) {
+      super(imagePb);
+      this.sourceDisk = DiskId.fromUrl(imagePb.getSourceDisk());
+      this.sourceDiskId = imagePb.getSourceDiskId();
+    }
+
+    /**
+     * Sets the identity of the source disk used to create the image.
+     */
+    public Builder sourceDisk(DiskId sourceDisk) {
+      this.sourceDisk = checkNotNull(sourceDisk);
+      return this;
+    }
+
+    Builder sourceDiskId(String sourceDiskId) {
+      this.sourceDiskId = sourceDiskId;
+      return this;
+    }
+
+    /**
+     * Creates a {@code DiskImageConfiguration} object.
+     */
+    @Override
+    public DiskImageConfiguration build() {
+      return new DiskImageConfiguration(this);
+    }
+  }
+
+  private DiskImageConfiguration(Builder builder) {
+    super(builder);
+    this.sourceDisk = checkNotNull(builder.sourceDisk);
+    this.sourceDiskId = builder.sourceDiskId;
+  }
+
+  /**
+   * Returns the identity of the source disk used to create this image.
+   */
+  public DiskId sourceDisk() {
+    return sourceDisk;
+  }
+
+  /**
+   * Returns the id of the disk used to create this image. This value may be used to determine
+   * whether the image was taken from the current or a previous instance of a given disk name.
+   */
+  public String sourceDiskId() {
```

Comment:
Oh, I am not sure why we deviated. I think because we have confusing terms complete/strict consistency with method name to expected return type is important. And therefore keeping `xxxId()` to always refer/return `XxxId`  and `id` to always refer and return the string generated id.

I think we should keep what you have as is *but* like @ajkannan I think we should find a way to distinct the 2 by property name. I think `serviceGeneratedId` is too long and would vote for `genratedId` (javadoc should mention that
this is a service generated id for the resource).